README.md

```diff
@@ -133,156 +189,219 @@ http://localhost:8092/content/remote/simple.html
+        },
+        {
+          "type": "remote",
+          "path": "/content/.*",
+          "domain": "localhost",
+          "port": 3001
+        }
+      ]
+    }
+  },
+  ...
+}
+```
+This section configures the Knot.x Repository Verticle listening for requests on Vert.x event bus. The config node consists of:
+- **service.name** - the name/address of the event bus to which Repository Verticle subscribes on,
+- **repositories** - an array of definitions of all repositories used that provide HTML Templates.
 
-### Services
+There are two sample repositories defined - `local` and `remote`. Each of them defines a `path` - a regular expression that indicates which resources will be taken from this repository. 
+The first one matched will handle the request or, if no repository is matched, **Knot.x** will return a `404 Not found` response for the given request.
 
-**service.yml**
-```yaml
-services:
+- **Local repositories**
+```json
+{
+    "type": "local",
+    "path": "/content/local/.*",
+    "catalogue": ""    
+}
+```
+If you need to take files from a local machine, this is the kind of repository you want to use. It's perfect for mocking data. 
+The second parameter to define is `catalogue` - it determines where to take the resources from. If it's left empty, they will be taken from the classpath. It may be treated like a prefix to the requested resources.
 
-  - path: /service/mock/.*
-    domain: localhost
-    port: 3000
+- **Remote repositories**
+This kind of repository connects with an external server to fetch templates. To specify where the remote instance is, please configure the `domain` and `port` parameters.
+```json
+{
+    "type": "remote",
+    "path": "/content/.*",
+    "domain": "localhost",
+    "port": 3001   
+}
+```
 
-  - path: /service/.*
-    domain: localhost
-    port: 8080
+**1.2. templateEngine** section
+```json
+  ...
+  "templateEngine": {
+    "config": {
+      "service.name": "template-engine",
+      "template.debug": true,
+      "services": [
+        {
+          "path": "/service/mock/.*",
+          "domain": "localhost",
+          "port": 3000
+        },
+        {
+          "path": "/service/.*",
+          "domain": "localhost",
+          "port": 8080
+        }
+      ]
+    }
+  },
+  ...,
 ```
+This section configures the Knot.x Template Engine Verticle listening for requests on Vert.x event bus. The config node consists of:
+- **service.name** - name/address of the event bus to which the Repository Verticle subscribes,
+- **template.debug** - boolean flag to enable/disable rendering HTML comment entities around dynamic snippets,
+- **services** - an array of definitions of all service endpoints used by dynamic snippets.
 
 There are two groups of services defined. Each one will be handled by a different server, i.e. all service requests which match the regular expression:
-
 - `/service/mock/.*` will by handled by `localhost:3000`
 - `/service/.*` will be handled by `localhost:8080`
```

Comment:
Do we want to provide configuration for external service in example configuration? In current situation it is not required.